config/coreboot-t440p.config

```diff
@@ -0,0 +1,18 @@
+# CONFIG_USE_BLOBS is not set
+CONFIG_VENDOR_LENOVO=y
+CONFIG_NO_POST=y
+CONFIG_CBFS_SIZE=0x800000
```

Comment:
I have the impression that your initial value might have been good, but not matching ifd defined regions.

Ifd needs to be modified to take output of neutered ME size, and the freed space added to BIOS region there. That would become the size of cbfs region under coreboot config from memory, but will have to find traces again which I was unsuccessful finding last time.

Will try to find that back.

Reply:
@rbreslow I finally joined coreboot channel and posted my question directly to them here: https://matrix.to/#/!BZxDFuoBnMKnzVZwEp:libera.chat/$N_Nh4qJeU9gquoE9tzjeERYzsxgYB6j79uB6uvnoPdM?via=libera.chat&via=matrix.org&via=foss.wtf

Reply:
> @rbreslow I finally joined coreboot channel and posted my question directly to them here: https://matrix.to/#/!BZxDFuoBnMKnzVZwEp:libera.chat/$N_Nh4qJeU9gquoE9tzjeERYzsxgYB6j79uB6uvnoPdM?via=libera.chat&via=matrix.org&via=foss.wtf

icon on coreboot channel answered. Excerpt:

> icon:
> it's not `end - start` but `end - start + 1`
> but that's not why it's slowing down. I assume with the bigger size, not everything in CBFS can be cached
> there's CAR (cache as RAM) at 0xff7c0000..0xff7effff. this is used to store data before the DRAM is up
> the BIOS region is mapped directly below 4G (0x100000000). so making CBFS too big, it could even conflict with CAR
> I don't think this can be changed without also changing the MRC blob
> so the best option is really to set CBFS_SIZE to 8MiB
> one could still make the rest of the BIOS region available with a custom FMAP. only not as part of CBFS

I replied:

> Interesting. Has no impact on ivy/sandy which extends cbfs region to reuse all liberated space from ME (11.5mb for heads maximized boards). But yet again, native ram init there.
> Misalignments increased boot time in the past, but those were linked to tpm measurements problems (increasing boot time by 50 seconds, but also on older versions of coreboot).

Then icon replied:

> yes, CAR is in 0xfefe0000..0xfefeffff for SNB
> that's DCACHE_RAM_BASE/_SIZE in Kconfig btw.
> oh, actually it's bigger

nic3-14159 added:

> Side note, IIRC the coreboot build system rounds CBFS_SIZE down to 64KiB boundaries

-----
@rbreslow Can you adjust the calculations and test with new CBFS? Might be a limitation of the MRC blob on which we might need to accept caching time (and document properly under board config as current limitation)

Reply:
> So maybe 0xBDEFFF is the maximum CBFS size? I will recompile, flash, and see. It would be cool if we could automate this math.

Basically, the right calculation is taking output of ifdtool -f layout.txt path/to/ifd_output_from_me_cleaner_on_fullrom_backup.rom and parsing the BIOS region, substracting end address to begin address.

@rbreslow Seems like xx30 and xx20 might have been off all along. Redoing.